### DISCUSSION TITLE
Logout route is displayed even if the session is inactive

### DISCUSSION THREAD

**Discussion by brunoprietog** [jeremyevans/rodauth#207] [General]
Hi,

That's basically what this is about. If my session is not logged in and I call /logout route, it should not show the view. In this case, I have the active_sesions feature enabled. Will that affect anything? Another thing to consider is that I am using rodauth-rails, but I don't think this is the cause of this behavior.

How could I make the path only show for authenticated users?

In my opinion, just like the other paths that automatically require login, this should also be one of them. I don't know if there are more routes that will have the same behavior.

Thanks!

**Comment by jeremyevans** [comment]
By default, the logout route is harmless on accounts that are logged in, since it only clears the session.  This is different than most other routes (e.g. change password), where you have to be logged in order for the route to make sense.  If for some reason you want to restrict views of the logout page for accounts that are not logged in, you could try:

```ruby
before_logout_route{require_login}
```

**Comment by brunoprietog** [comment]
Thanks!
